src/redux/features/posts/postsSlice.js

```diff
@@ -23,7 +45,111 @@ export const postsSlice = createSlice({
         postsAPI.endpoints.getPosts.matchRejected,
         (state, action) => {
           state.status = "error";
-          state.error = action.payload.error;
+          state.error = action.payload.data.error;
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.postLikeCall.matchPending,
+        actionLikePending
+      )
+      .addMatcher(
+        postsAPI.endpoints.postLikeCall.matchFulfilled,
+        actionLikeFulfilled
+      )
+      .addMatcher(
+        postsAPI.endpoints.postLikeCall.matchRejected,
+        actionLikeRejected
+      )
+      .addMatcher(
+        postsAPI.endpoints.postDisLikeCall.matchPending,
+        actionLikePending
+      )
+      .addMatcher(
+        postsAPI.endpoints.postDisLikeCall.matchFulfilled,
+        actionLikeFulfilled
+      )
+      .addMatcher(
+        postsAPI.endpoints.postDisLikeCall.matchRejected,
+        actionLikeRejected
+      )
+      .addMatcher(
+        postsAPI.endpoints.deleteBroadcastCall.matchPending,
+        (state) => {
+          state.deleteStatus = "loading";
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.deleteBroadcastCall.matchFulfilled,
+        (state, action) => {
+          state.deleteStatus = "success";
+          state.data = action.payload.posts;
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.deleteBroadcastCall.matchRejected,
+        (state, action) => {
+          state.deleteStatus = "error";
+          state.error = action.payload.data.error;
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.postNewBroadcastCall.matchPending,
+        (state) => {
+          state.newBroadcastStatus = "loading";
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.postNewBroadcastCall.matchFulfilled,
+        (state, action) => {
+          state.newBroadcastStatus = "success";
+          state.data = action.payload.posts;
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.postNewBroadcastCall.matchRejected,
+        (state, action) => {
+          state.newBroadcastStatus = "error";
+          state.error = action.payload.data.error;
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.postEditedBroadcastCall.matchPending,
+        (state) => {
+          state.editBroadcastStatus = "loading";
+        }
+      )
+      .addMatcher(
+        postsAPI.endpoints.postEditedBroadcastCall.matchFulfilled,
+        (state, action) => {
+          state.editBroadcastStatus = "success";
```

Comment:
You can use constants here in place of success

Reply:
Thanks for suggesting, Will add in next PR.